scene/gui/line_edit.cpp

```diff
@@ -633,8 +633,8 @@ void LineEdit::_notification(int p_what) {
 				if (char_ofs >= t.length())
 					break;
 
-				CharType cchar = pass ? '*' : t[char_ofs];
-				CharType next = pass ? '*' : t[char_ofs + 1];
+				CharType cchar = (pass && !text.empty()) ? t[char_ofs] : '*';
+				CharType next = (pass && !text.empty()) ? t[char_ofs + 1] : '*';
```

Comment:
I think your check is wrong. Currently, this would show the actual characters only if pass is true and text is not empty. Maybe you want to invert the positions of the `'*'` and `t[char_ofs...]`?